pkg/kp/pcstore/consul_store_test.go

```diff
@@ -211,6 +212,49 @@ func TestDelete(t *testing.T) {
 	}
 }
 
+func TestWatch(t *testing.T) {
+	store := consulStoreWithFakeKV()
+	pod := fields.ID("pod_id")
+	podID := types.PodID("pod_id")
+	az := fields.AvailabilityZone("us-west")
+	clusterName := fields.ClusterName("cluster_name")
+
+	selector := klabels.Everything().
+		Add(fields.PodIDLabel, klabels.EqualsOperator, []string{pod.String()}).
+		Add(fields.AvailabilityZoneLabel, klabels.EqualsOperator, []string{az.String()}).
+		Add(fields.ClusterNameLabel, klabels.EqualsOperator, []string{clusterName.String()})
+
+	annotations := fields.Annotations(map[string]interface{}{
+		"foo": "bar",
+	})
+
+	quit := make(chan struct{})
+	watch := store.Watch(pod, quit)
+
+	var watched []WatchedPodCluster
```

Comment:
looks like you only attempt to put one value in here, should this be a pointer instead?